### PR TITLE
Fat 5587 create cartforbuy

### PR DESCRIPTION
**Pull request by Activict**

### CODE REVIEW

File: StoreApp/Controllers/CartController.cs [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+            if (TempData.Peek("Сart") is List<ProductViewModel>)
+            {
+                var cart = TempData.Peek("Сart") as List<ProductViewModel>;
+
+                foreach (var product in cart)
+                {
+                    ViewBag.CountToCart += product.Quantity;
+                    ViewBag.TotalPriceCart += product.Quantity * product.Price;
+                }
```
Comment (AleksandroBoko) [inline StoreApp/Controllers/CartController.cs]: Try to use LINQ to optimize the code

File: StoreApp/Controllers/CartController.cs [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+                for (int i = 0; i < cart.Count; i++)
+                {
+                    if (cart[i].Id == id)
+                    {
+                        cart.RemoveAt(i);
+
+                        TempData["Cart"] = cart;
+                    }
+                }
```
Comment (AleksandroBoko) [inline StoreApp/Controllers/CartController.cs]: Also you can try to use LINQ

File: StoreApp/Controllers/CartController.cs [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+                for (int i = 0; i < cart.Count; i++)
+                {
+                    if (cart[i].Id == id)
+                    {
+                        cart[i].Quantity++;
+
+                        TempData["Cart"] = cart;
+                    }
+                }
```
Comment (AleksandroBoko) [inline StoreApp/Controllers/CartController.cs]: Try to use LINQ to optimize the code

File: StoreApp/Controllers/CartController.cs [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+                            cart[i].Quantity--;
+                        }
+                        else
+                        {
+                            cart.RemoveAt(i);
+                        }
+                        
+                        TempData["Cart"] = cart;
+                    }
+                }
```
Comment (AleksandroBoko) [inline StoreApp/Controllers/CartController.cs]: Try to use LINQ to optimize the code

File: StoreApp/Controllers/CartController.cs [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+            if (TempData.Peek("Сart") is List<ProductViewModel>)
+            {
+                var cart = TempData.Peek("Сart") as List<ProductViewModel>;
+
+                decimal totalPriceCart = 0;
+
+                foreach (var product in cart)
+                {
+                    totalPriceCart += product.Quantity * product.Price;
+                }
```
Comment (AleksandroBoko) [inline StoreApp/Controllers/CartController.cs]: Try to use LINQ to optimize the code

File: StoreApp/Controllers/CartController.cs [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+
+                int orderId = -1;
```
Comment (AleksandroBoko) [inline StoreApp/Controllers/CartController.cs]: Do you need to set this value?

File: StoreApp/Controllers/CartController.cs [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+
+                if (cart.Count > 0 && totalPriceCart > 0)
+                {
+                    string username;
+                    if (User.Identity.Name == "") username = "user1"; 
+                    else username = User.Identity.Name;
+
+                    int userId = userService.GetAll().First(u => u.UserName.Equals(username)).Id;
```
Comment (AleksandroBoko) [inline StoreApp/Controllers/CartController.cs]: 1. Style of **if block** is not correct.
2. As I could understand it's a temporary solution?

File: StoreApp/Controllers/CartController.cs [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+
+                    orderId = orderService.Create(userId, totalPriceCart);
+                }
+                else
+                {
+                    TempData["Message"] = "Order don't created. Cart is empty";
```
Comment (AleksandroBoko) [inline StoreApp/Controllers/CartController.cs]: Just a grammar mistake. It's right to write - **The order wasn't created**

File: StoreApp/Controllers/CartController.cs [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+                    OrderDetailDTO orderDetailDTO = new OrderDetailDTO()
+                    {
+                        OrderId = orderId,
+                        Price = product.Price,
+                        ProductId = product.Id,
+                        Quantity = product.Quantity
+                    };
+
+                    orderDetailService.Create(orderDetailDTO);
+                }
```
Comment (AleksandroBoko) [inline StoreApp/Controllers/CartController.cs]: I suppose we can move this block of code into a new method or orderDetailService

File: StoreApp/Controllers/CartController.cs [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+
+                TempData["Сart"] = null;
+                TempData["Message"] = "Order created success";
+
+                return RedirectToAction("Index");
+            }
+            else
+            {
+                TempData["Message"] = "Order don't created. Cart is empty";
```
Comment (AleksandroBoko) [inline StoreApp/Controllers/CartController.cs]: Also the same mistake

File: StoreApp/Controllers/StoreController.cs [hunk 11]
```diff
@@ -199,6 +200,37 @@ public ActionResult DeleteProduct(ProductViewModel product)
...
         }
 
+        [HttpGet]
+        public ActionResult Buy(int id)
+        {
+            ProductViewModel productVM = webMapper.config.Map<ProductDTO, ProductViewModel>(productService.Get(id));
+
+            if (productVM == null)
+            {
+                TempData["Message"] = "Product don't add to cart";
```
Comment (AleksandroBoko) [inline StoreApp/Controllers/StoreController.cs]: It's right to say **"The product wasn't added to the cart"**

File: StoreApp/Controllers/StoreController.cs [hunk 12]
```diff
@@ -199,6 +200,37 @@ public ActionResult DeleteProduct(ProductViewModel product)
...
+            }
+            else
+            {
+                productVM.Quantity = 1;
+                cart.Add(productVM);
+            }
+
+            TempData["Сart"] = cart;
+
+            TempData["Message"] = "Product added to cart";
```
Comment (AleksandroBoko) [inline StoreApp/Controllers/StoreController.cs]: Grammar mistake **"The product was added to the cart"**

File: StoreApp/Views/Store/DetailsProduct.cshtml [hunk 13]
```diff
@@ -92,7 +92,9 @@
 
     </dl>
 </div>
+
```
Comment (AleksandroBoko) [inline StoreApp/Views/Store/DetailsProduct.cshtml]: My proposition to remove empty space between tags.

File: StoreApp/Controllers/CartController.cs (hunk 1 above)
Comment (Activict) [reply StoreApp/Controllers/CartController.cs]: fixed

File: StoreApp/Controllers/CartController.cs (hunk 8 above)
Comment (Activict) [reply StoreApp/Controllers/CartController.cs]: fixed

File: StoreApp/Controllers/CartController.cs (hunk 10 above)
Comment (Activict) [reply StoreApp/Controllers/CartController.cs]: fixed

File: StoreApp/Controllers/StoreController.cs (hunk 11 above)
Comment (Activict) [reply StoreApp/Controllers/StoreController.cs]: fixed

File: StoreApp/Controllers/StoreController.cs (hunk 12 above)
Comment (Activict) [reply StoreApp/Controllers/StoreController.cs]: fixed

File: StoreApp/Views/Store/DetailsProduct.cshtml (hunk 13 above)
Comment (Activict) [reply StoreApp/Views/Store/DetailsProduct.cshtml]: fixed

File: StoreApp/Controllers/CartController.cs (hunk 2 above)
Comment (Activict) [reply StoreApp/Controllers/CartController.cs]: fixed

File: StoreApp/Controllers/CartController.cs (hunk 3 above)
Comment (Activict) [reply StoreApp/Controllers/CartController.cs]: fixed

File: StoreApp/Controllers/CartController.cs (hunk 4 above)
Comment (Activict) [reply StoreApp/Controllers/CartController.cs]: fixed

File: StoreApp/Controllers/CartController.cs (hunk 5 above)
Comment (Activict) [reply StoreApp/Controllers/CartController.cs]: fixed

**Pull request merged**